source/sigv4.c

```diff
@@ -2338,12 +2393,12 @@ static int32_t hmacFinal( HmacContext_t * pHmacContext,
     if( returnStatus == 0 )
     {
         /* Create the outer-padded key by retrieving the original key from
-         * the inner-padded key then XOR with opad. XOR is associative,
-         * so one way to do this is by performing XOR on each byte of the
-         * inner-padded key with (0x36 ^ 0x5c) = (ipad ^ opad) = 0x6a.  */
+         * the inner-padded key (by XORing with ipad byte) and then XOR with opad
+         * to generate the outer-padded key. As XOR is associative, one way to do
+         * this is by performing XOR on each byte of the inner-padded key (ipad ^ opad).  */
         for( i = 0U; i < pCryptoInterface->hashBlockLen; i++ )
         {
-            pHmacContext->key[ i ] ^= 0x6aU;
+            pHmacContext->key[ i ] ^= ( HMAC_INNER_PAD_BYTE ^ HMAC_OUTER_PAD_BYTE );
```

Comment:
This may XOR ipad and opad constants on each iteration depending on the compiler. Better to have a macro for `0x6aU`.